packages/ui-simple-select/src/SimpleSelect/index.tsx

```diff
@@ -169,10 +163,14 @@ class SimpleSelect extends Component<SimpleSelectProps, SimpleSelectState> {
   }
 
   getFirstOption() {
+    const children = Children.toArray(this.props.children) as (
+      | OptionChild
+      | GroupChild
+    )[]
     let match: OptionChild | undefined
 
-    for (let i = 0; i < this.childrenArray.length; i++) {
-      const child = this.childrenArray[i]
```

Comment:
this was calling the getter in every iteration, I can imagine that it causes a slowdown for lots of elements.